src/control/server/ctl_smd_rpc.go

```diff
@@ -382,18 +498,14 @@ func (svc *ControlService) SmdQuery(ctx context.Context, req *ctlpb.SmdQueryReq)
 		return nil, FaultDataPlaneNotStarted
 	}
 
-	if req.SetFaulty {
-		return svc.smdSetFaulty(ctx, req)
+	if req.SetFaulty || req.Identify || req.ResetLed || req.GetLed {
+		return svc.smdManage(ctx, req)
```

Comment:
Seems like a good argument for breaking the commands off into their own function at this level, too.

Reply:
I think I've already tried to explain the rationale for this in previous comment replies.